### PR TITLE
fix(agents): restore max turns on agent harness

### PR DESCRIPTION
**Pull request by jordan-umusu**
## Summary by cubic
Restores max-turn enforcement in the agent harness end-to-end to prevent unbounded loops and honor the configured workflow limit.

- **Bug Fixes**
  - Propagate `max_requests` from durable workflow to executor and approval continuations.
  - Add `max_requests` to `AgentExecutorInput`, `LoopbackInput`, and `RuntimeInitPayload` (serialize/deserialize); map to Claude runtime `max_turns`.
  - Add unit test ensuring `max_requests` is forwarded as `max_turns` (covers None and explicit limits).

<sup>Written for commit 48e49405d3ee8182311703995bf360ddf77f6726. Summary will update on new commits.</sup>

### CODE REVIEW

**Review by daryllimyt — APPROVED** [review]
lgtm, add small test

File: packages/tracecat-ee/tracecat_ee/agent/workflows/durable.py [hunk 1]
```diff
@@ -591,6 +592,7 @@ async def _run_with_nsjail(
                     sdk_session_data=self._sdk_session_data,
                     is_approval_continuation=True,
                     use_workspace_credentials=args.agent_args.use_workspace_credentials,
+                    max_requests=self.max_requests,
```
[the remark this answers is not in the document]
Comment (jordan-umusu) [reply packages/tracecat-ee/tracecat_ee/agent/workflows/durable.py:595]: Agreed but non-trivial and not worth implementing at this time.
